src/blob.jl

```diff
@@ -88,16 +106,16 @@ end
 
 @generated function Base.getindex(blob::Blob{T}, ::Type{Val{field}}) where {T, field}
     i = findfirst(isequal(field), fieldnames(T))
-    @assert i != nothing "$T has no field $field"
+    @assert i !== nothing "$T has no field $field"
     quote
         $(Expr(:meta, :inline))
-        Blob{$(fieldtype(T, i))}(blob + $(blob_offset(T, i)))
+        $(Blob{fieldtype(T, i)})(blob, $(blob_offset(T, i)))
```

Comment:
I _think_ / hope that the size of the code generated shouldn't be affected at all by moving the Blob{T} type construction inside or outside the quote, since it should definitely compile away during Julia's inference phase, and shouldn't show up at all in the LLVM code?
But i think it's fine how you have it too 👍